pkg/descheduler/controllers/migration/reservation/util_test.go

```diff
@@ -0,0 +1,192 @@
+										RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+											NodeSelectorTerms: []corev1.NodeSelectorTerm{
+												{
+													MatchFields: []corev1.NodeSelectorRequirement{
+														{
+															Key:      "metadata.name",
+															Operator: corev1.NodeSelectorOpNotIn,
+															Values:   []string{"test-node"},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "skip current node with existing node affinity",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					OwnerReferences: ownerReferences,
+					Namespace:       "default",
+				},
+				Spec: corev1.PodSpec{
+					NodeName: "test-node",
+					Affinity: &corev1.Affinity{
+						NodeAffinity: &corev1.NodeAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+								NodeSelectorTerms: []corev1.NodeSelectorTerm{
+									{
+										MatchExpressions: []corev1.NodeSelectorRequirement{
+											{
+												Key:      "test",
+												Operator: corev1.NodeSelectorOpIn,
+												Values:   []string{"xxxx"},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOptions: &sev1alpha1.PodMigrateReservationOptions{
+				Template: &sev1alpha1.ReservationTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							LabelCreatedBy: "koord-descheduler",
+						},
+					},
+					Spec: sev1alpha1.ReservationSpec{
+						AllocateOnce: true,
+						Owners: []sev1alpha1.ReservationOwner{
+							{
+								Controller: &sev1alpha1.ReservationControllerReference{
+									OwnerReference: ownerReferences[0],
+									Namespace:      "default",
+								},
+							},
+						},
+						Template: &corev1.PodTemplateSpec{
+							ObjectMeta: metav1.ObjectMeta{
+								Namespace:       "default",
+								OwnerReferences: ownerReferences,
+							},
+							Spec: corev1.PodSpec{
+								Affinity: &corev1.Affinity{
+									NodeAffinity: &corev1.NodeAffinity{
+										RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+											NodeSelectorTerms: []corev1.NodeSelectorTerm{
+												{
+													MatchExpressions: []corev1.NodeSelectorRequirement{
+														{
+															Key:      "test",
+															Operator: corev1.NodeSelectorOpIn,
+															Values:   []string{"xxxx"},
+														},
+													},
+													MatchFields: []corev1.NodeSelectorRequirement{
+														{
+															Key:      "metadata.name",
+															Operator: corev1.NodeSelectorOpNotIn,
+															Values:   []string{"test-node"},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Maybe we should check whether UpdatedReservationOptions works by getRequiredNodeAffnity and matching node.